hazelcast/src/main/java/com/hazelcast/client/HazelcastClient.java

```diff
@@ -439,46 +443,51 @@ private static HazelcastInstance getOrCreateClientInternal(ClientConfig config)
         }
 
         try {
-            return constructHazelcastClient(null, config, null, instanceName, future);
+            return constructHazelcastClient(instanceName, config, null, null, null, future);
         } catch (Throwable t) {
             CLIENTS.remove(instanceName, future);
             future.setFailure(t);
             throw ExceptionUtil.rethrow(t);
         }
     }
 
-    private static HazelcastInstance constructHazelcastClient(AddressProvider addressProvider, ClientConfig clientConfig,
-                                                              ClientFailoverConfig failoverConfig, String instanceName,
-                                                              InstanceFuture<HazelcastClientProxy> future) {
+    private static HazelcastInstance constructHazelcastClient(
+            String instanceName,
+            ClientConfig config,
+            ClientFailoverConfig failoverConfig,
+            ClientConnectionManagerFactory connectionManagerFactory,
+            AddressProvider addressProvider,
+            InstanceFuture<HazelcastClientProxy> future
+    ) {
         final ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
         HazelcastClientProxy proxy;
         try {
             Thread.currentThread().setContextClassLoader(HazelcastClient.class.getClassLoader());
-            ClientConnectionManagerFactory factory = new DefaultClientConnectionManagerFactory();
-            HazelcastClientInstanceImpl client = new HazelcastClientInstanceImpl(instanceName, clientConfig,
-                    failoverConfig, factory, addressProvider);
+            if (connectionManagerFactory == null) {
+                connectionManagerFactory = new DefaultClientConnectionManagerFactory();
+            }
+            HazelcastClientInstanceImpl client = new HazelcastClientInstanceImpl(instanceName, config,
+                    failoverConfig, connectionManagerFactory, addressProvider);
             client.start();
             OutOfMemoryErrorDispatcher.registerClient(client);
             proxy = new HazelcastClientProxy(client);
             future.set(proxy);
-        } catch (Throwable t) {
```

Comment:
What do we achieve by this change?

Reply:
It was already being caught and rethrown on the codes that use this function. Removed this one to get rid of this unnecessary step